389 - Call for contributions: Guidelines and Banner

nozomione opened the pull request:
## Issue Number
#389

## Purpose/Implementation Notes
<!-- For changes that impact the frontend and user interactions, please ensure there's a vercel preview and tag a designer (@dvenprasad) for review  -->

Added the contribution guidelines banner and included the necessary buttons in md file.

The following pages are excluded from the banner display:
(*If the banner needs to be shown on those pages, please let me know)
- [Contribute](https://scpca-portal-ev6izpyko-ccdl.vercel.app/contribute)
- [Privacy Policy](https://scpca-portal-ev6izpyko-ccdl.vercel.app/privacy-policy)
- [Terms of use](https://scpca-portal-ev6izpyko-ccdl.vercel.app/terms-of-use)

- [ ] UI review w/ @dvenprasad  - please visit the latest [here](https://scpca-portal-pek9s2kgn-ccdl.vercel.app), thank you!

## Types of changes

What types of changes does your code introduce?

<!-- Remove any which your PR isn't -->

- New feature (non-breaking change which adds functionality)

## Functional tests

List out the functional tests you've completed to verify your changes work locally.

## Checklist
- [x] Lint and unit tests pass locally with my changes

dvenprasad reviewed (commented):
The banner and the contribute page are looking good!

Two comments:
- On Browse and Experiment page, the spacing between the banner and the rest of the page is too much. Can we bring it down to 40 px, please?

- On the contribute page, please add a `Fill the Inkate Form` button at the bottom of the page too.

davidsmejia commented on client/src/components/Layout.js:
```diff
@@ -29,8 +30,19 @@ const ProgressBar = styled(PageLoader)`
 export const Layout = ({ children }) => {
-  const { bannerHeight } = useBanner()
   const router = useRouter()
+  const { stickyBannerHeight } = useStickyBanner()
```
I suggest that we instead:
- track updates to fixed header on open/close of sticky banner
- move contribute banner inside of main
- add support for named banners
- have only one banner that takes children to render contents, a name and a flag for fullWidth
- move the nf / contribute content into named banner components ie `ContributeBanner.js` and pass that as a child into `Banner`

nozomione replied:
Hello @davidsmejia, I updated the following items and now the code is ready for your review 👀 
>     * track updates to fixed header on open/close of sticky banner
Instead, I incorporated the [`ResizeObserver`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver) Web API(as a hook - see the resource included in the comment). It let us track DOM element object changes with no additional checks and also circumvent the `useRef` limitation to detect DOM changes

>     * move contribute banner inside of main
I discovered that our current structure's `main` is set to fixed width (not full), and the banner needs to be extended as full width, so it remained in place (otherwise we could u[date the site structure, but that could require another PR!)

>     * add support for named banners
>     * have only one banner that takes children to render contents, a name and a flag for fullWidth
>     * move the nf / contribute content into named banner components ie `ContributeBanner.js` and pass that as a child into `Banner`
Now it takes the unique`id` and `fullWidth` props and takes `children`, and created named banner components.

davidsmejia commented on client/src/config/contribution-guidelines.md:
```diff
@@ -25,6 +29,10 @@ We will accept submissions of 10x Genomics single-cell or single-nuclei profilin
 Researchers must process their data using the Childhood Cancer Data Lab's production pipeline – available at <https://github.com/AlexsLemonade/scpca-nf/> – and submit the output, along with project (see [**Submitting Project Metadata**](#submitting-project-metadata)), sample (see [**Submitting Sample Metadata**](#submitting-sample-metadata)) and cell (see [**Submitting Cell Metadata**](#submittin ...
 Submitters must also sign a Data Transfer Agreement before data transfer ([view Data Transfer Agreement template](https://docs.google.com/document/d/1eLHobiV4M0bC0KOQALgpfaLU0Y4nMZYs/edit?usp=share_link&ouid=105890053693989014850&rtpof=true&sd=true)).
 
+<form>
```
Let's replace this with either a comment or a link and ask Jackie to update the main copy.

If using a comment:
- do a normal string replace with the form html that below this comment

If using a link:
- override the createElement and find the link you want to replace with a button

davidsmejia commented on client/src/pages/contribute.js:
```diff
@@ -0,0 +1,149 @@
+const headingMargin = { top: '24px', bottom: 'small' }
+const listMargin = { bottom: 'medium', horizontal: 'medium' }
+
+export const Contribute = () => {
```
Swap the a: with Link to handle external links opening in a new page.

davidsmejia reviewed (changes requested):
Looking good some structural changes requested

nozomione commented:
>  * On Browse and Experiment page, the spacing between the banner and the rest of the page is too much. Can we bring it down to 40 px, please?
>   * On the contribute page, please add a `Fill the Inkate Form` button at the bottom of the page too.

@dvenprasad  I updated the above items and the UI is now ready for your review [here](https://scpca-portal-ej9xk5qse-ccdl.vercel.app), thank you!

nozomione commented:
@jaclyn-taroni Please edit the md file and insert the comment `<!-- IntakeFormLink -->` at line 28 and 201 (at the bottom of the file) and add additional lines around it ([reference](https://github.com/AlexsLemonade/scpca-portal/pull/398/files#diff-f8f901802c9dc2ad11f47cd76d967fc69862b67c9e780a6400b6f6c6c07691f4)), thank you!

**NOTE:** Please disregard the formatting done by the editor's markdown plugin in the reference file.

dvenprasad reviewed (approved):
Looks good! 
🏇 🏇 🏇

davidsmejia reviewed (changes requested):
A couple things we can clean up here:
- fixedBoxHeight instead of margin
- Change name of `ContributeBanner` to `ContributeContentBox` and remove Banner logic from that component
- Negate props for Banner component

nozomione commented:
@davidsmejia Thank you for the great suggestion! 
> 
>     * fixedBoxHeight instead of margin
Removed the `bottomMargin` logic ✅ 
 
>     * Change name of `ContributeBanner` to `ContributeContentBox` and remove Banner logic from that component
For this suggestion, I revised the named components and extract each content to be an exportable sub-component (so it can be used anywhere in our codebase independently as needed). In addition, they now take the following props:
- `id`(with its default value) so that id can be programmatically set
- props object and pass it to the Banner

>     * Negate props for Banner component
Thank you for this great suggestion ✅ ! 
Due to this update, I now added the `<RecruitNFBanner hidden />` in `Layout` and we can simply toggle it as needed in the future!)

davidsmejia reviewed (approved):
LGTM!
